app/models/concerns/user/omniauth.rb

```diff
@@ -9,16 +9,23 @@ def add_data_from_facebook(auth)
 
   class_methods do
     def from_omniauth(auth)
-      user = where(provider: auth.provider, uid: auth.uid).first_or_initialize do |user|
+      find_by(provider: auth.provider, uid: auth.uid) ||
+        find_by(email: auth.info.email) ||
+        create_from_omniauth(auth)
+    end
+
+    def create_from_omniauth(auth)
+      User.new.tap do |user|
+        user.provider = auth.provider
+        user.uid = auth.uid
         user.email = auth.info.email
         user.password = Devise.friendly_token[0, 20]
         user.skip_confirmation!
+
+        user.profile.name = [auth.info.first_name, auth.info.last_name].join(' ')
+        user.profile.userpic = URI.parse(auth.info.image).open if auth.info.image.present?
```

Comment:
As far as I remember, there was a problem creating userpic from inside block where user isn't initialized yet.
Can you test it?

Reply:
I'll do. Thank you for pointing on it.